src/mpid/ch4/netmod/ofi/ofi_impl.h

```diff
@@ -115,9 +115,10 @@
 
 #define MPIDI_OFI_CALL_LOCK 1
 #define MPIDI_OFI_CALL_NO_LOCK 0
-#define MPIDI_OFI_CALL_RETRY(FUNC,STR,LOCK)                               \
+#define MPIDI_OFI_CALL_RETRY(FUNC,STR,LOCK,EAGAIN)          \
     do {                                                    \
     ssize_t _ret;                                           \
+    int _retry = EAGAIN ? 1 : -1;                           \
```

Comment:
Hmm, so this will add one branch in the critical path... how about making it configure-time option to enable/disable until `MPI_ERR_EAGAIN` becomes mandatory?

Reply:
Configure check seems like a really heavy hammer for this?  How about changing the caller to add the retry count instead?  We'd get that count from a CVAR, which would default to `-1`.

Reply:
I have added the CVAR and moved the check out of the loop. So in the case where EAGAIN set to false, the retry loop won't pay the cost of one extra branch.

Reply:
The conditional here is evaluating compile-time constant. It shouldn't actually be a penalty. That said, I agree that a CVAR is a good idea.

Reply:
Not quite. Because for send, the EAGAIN is from the info hint of the communicator, which can only be evaluated at runtime. But I think the major concern was that we were checking EAGAIN in the following loop (which we no longer do in the latest patch).

Reply:
Oh, right. I was thinking of a previous version. Sorry about that.

Reply:
:+1: Current version of the patch looks good. Thanks!